actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -144,11 +146,22 @@ def self.const_missing(const_name)
     #   params = ActionController::Parameters.new(name: 'Francesco')
     #   params.permitted?  # => true
     #   Person.new(params) # => #<Person id: nil, name: "Francesco">
-    def initialize(attributes = nil)
-      super(attributes)
+    def initialize(parameters = {})
+      @parameters = parameters.with_indifferent_access
```

Comment:
I tried to change the constructor to not calling `.with_indifferent_access` but that totally broke the tests. I think it's better to keep it like that to make sure that params we are dealing with is HWIA. I feel like we'll run into weird issue if we don't do that.